apple/internal/partials/main_thread_checker_dylibs.bzl

```diff
@@ -0,0 +1,122 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Partial implementation for Main Thread Checker libraries processing."""
+
+load(
+    "@build_bazel_rules_apple//apple/internal:intermediates.bzl",
+    "intermediates",
+)
+load(
+    "@build_bazel_rules_apple//apple/internal:processor.bzl",
+    "processor",
+)
+load(
+    "@build_bazel_rules_apple//apple/internal/utils:main_thread_checker_dylibs.bzl",
+    "main_thread_checker_dylibs",
+)
+load(
+    "@build_bazel_apple_support//lib:apple_support.bzl",
+    "apple_support",
+)
+load(
+    "@bazel_skylib//lib:partial.bzl",
+    "partial",
+)
+
+def _main_thread_checker_dylibs_partial_impl(
+        *,
+        actions,
+        apple_mac_toolchain_info,
+        binary_artifact,
+        features,
+        label_name,
+        output_discriminator,
+        platform_prerequisites,
+        dylibs):
+    """Implementation for the Main Thread Checker dylibs processing partial."""
+    bundle_zips = []
+    if main_thread_checker_dylibs.should_package_main_thread_checker_dylib(features = features):
+        main_thread_checker_zip = intermediates.file(
```

Comment:
Is this is a zip to support a future case where more than one dylib is applied at the same time (e.g., the various sanitizers)?  If we know it will only ever be one can be return it directly and skip the zip?

Reply:
Oh yes, I initially copied it from clang sanitizer's related code :) Updated to only return the dylib and skip the zipping...